common/src/chain/transaction/signed_transaction.rs

```diff
@@ -92,11 +88,6 @@ impl SignedTransaction {
     }
 }
 
-// TODO enforce that inputs size is equal to signatures size when decoding
-
-// TODO make the SignedTransaction serialization ignore the size of the witness vec and just use the size of the inputs
-// NOTE: there might be difficulties there as Encode cannot fail. It may lead to accepting a panic there
```

Comment:
I think it would be nice if the `SignedTransaction` type maintained the invariant internally. Allows for early rejection of obviously malformed transactions before hitting `chainstate`.

Reply:
I'm torn here... I'm hesitating to do this decode thing because I'm afraid we'd modify it in the future and forget to update the Decode impl. But maybe I should just get over myself and do it... not sure...

Reply:
Done.